gpbft/chain.go

```diff
@@ -164,6 +164,22 @@ func (c ECChain) HasPrefix(other ECChain) bool {
 	return true
 }
 
+// Checks whether a chain has some prefix (including the head).
+// Always false for a zero value.
+func (c ECChain) HasPrefixAssumingCorrectChains(other ECChain) bool {
```

Comment:
I can't figure out what this is doing. Is it supposed to have the same semantics as HasPrefix but be an optimisation? What is "correct chains"?

Reply:
The other function that is written checks every tipset, but that is an overkill if chains are correct (i.e. valid): as soon as the head of one is contained in the chain of the other we can already conclude that they are prefixes, no need to check for all the parents following up.

We can probably just have this function because it seems weird to me that an ECChain that is not correct, but thought that it can be made in a different PR not to make too many things in this one.

Reply:
I just removed this function following your other comment.